### PR TITLE
refactor(test): rename and add scenario for test-stack-prevent-native-dismiss-single-stack

### PR DESCRIPTION
**Pull request by LKuchno**
## Description

Closes: https://github.com/software-mansion/react-native-screens-labs/issues/1614

Adds a manual test scenario (`scenario.md`) for the gamma/v5 Stack `preventNativeDismiss` single-stack test screen, and renames the screen directory to follow the repo's `test-*` naming convention.

The scenario documents the Android-only `preventNativeDismiss` behavior across both flag states:

- **Enabled** — native dismissal (native header back-button chevron and the system gesture-back / edge swipe) is intercepted, `onNativeDismissPrevented` fires, and a green "Native dismiss prevented" toast is shown; the screen stays put.
- **Disabled** — the same native gestures pop the stack normally, with no toast.
- The on-screen **Pop** button is JS-driven and always pops regardless of the flag.
- The **Toggle Prevent Native Dismiss** button flips the flag at runtime (via `setRouteOptions`), and the scenario verifies interception follows the current value.

To exercise the native-back / gesture-back flows on Android, the scenario uses the **direct launch** approach: the screen is rendered as the root component via `apps/App.tsx` (`TestStackPreventNativeDismissSingleStack as Example`). With the gamma `StackContainer` at the root, native back and gesture-back interact with the stack directly, which works around [#1459](https://github.com/software-mansion/react-native-screens-labs/issues/1459) (where, when the screen is nested inside the example app's own navigation, native back navigates out to the system menu instead of popping).

## Changelog

- Rename `prevent-native-dismiss-single-stack/` → `test-stack-prevent-native-dismiss-single-stack/` (aligns with the `test-*` convention) and update its registration in `stack-v5/index.ts`.
- Add `scenario.md` documenting the Android-only prevent-native-dismiss single-stack manual test: direct-launch Android setup, enabled-state interception (header chevron + gesture-back), disabled-state native-back pop, runtime flag toggling, the always-works on-screen **Pop**, and an edge case for toggling immediately before a native dismiss. The [#1459](https://github.com/software-mansion/react-native-screens-labs/issues/1459) direct-launch workaround is documented in the scenario's Note.

### CODE REVIEW

File: apps/src/tests/single-feature-tests/stack-v5/test-stack-prevent-native-dismiss-single-stack/scenario.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+  **Settings → Navigation mode → select Gesture navigation**.
+
+## Note
+
+- iOS does not yet support `preventNativeDismiss` for stack v5 - it is unimplemented.
+- The `Key` has the form `r-<routeName>-<n>`, where `<n>` comes from a
+  session-global counter that increments on every push and is **never
+  reset**. Do **not** expect specific numbers such as `r-A-1`; only the
+  **relationships** matter — every push produces a strictly new (higher)
+  `Key`, and a preserved (not recreated) screen keeps the same `Key`.
```
Comment (kkafar) [inline apps/src/tests/single-feature-tests/stack-v5/test-stack-prevent-native-dismiss-single-stack/scenario.md]: I don't know yet whether this is relevant or not, but this is an implementation detail we should avoid relying upon in tests. It's not guaranteed to look this way. Might be changed w/o any notice.

File: apps/src/tests/single-feature-tests/stack-v5/test-stack-prevent-native-dismiss-single-stack/scenario.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+- The on-screen **Pop** button always pops regardless of the flag.
+
+## Steps
+
+### Baseline
+
+1. Launch the app directly via `App.tsx` so the
+   **Home** screen is shown.
+
+- [ ] **Home** is shown (light blue background), `Name: Home` and a `Key`
```
Comment (kkafar) [inline apps/src/tests/single-feature-tests/stack-v5/test-stack-prevent-native-dismiss-single-stack/scenario.md]:
```suggestion
- [ ] **Home** is shown, `Name: Home` and a `Key`
```

Redundant detail -> the background color might be easily changed w/o any notice and this will go out of date.

**Review by kkafar — APPROVED** [review]
Looks good. I just have some minor remarks. Resolve them & land the PR.

Thank you.

File: apps/src/tests/single-feature-tests/stack-v5/test-stack-prevent-native-dismiss-single-stack/scenario.md (hunk 2 above)
Comment (LKuchno) [reply apps/src/tests/single-feature-tests/stack-v5/test-stack-prevent-native-dismiss-single-stack/scenario.md]: Scenario updated:
https://github.com/software-mansion/react-native-screens/pull/4272/changes/ee804f04140a88cdaa167a4d6336650c69df87a6
https://github.com/software-mansion/react-native-screens/pull/4272/changes/292f0b6de212a5d743c76d07f835a2e7571cfa9f

File: apps/src/tests/single-feature-tests/stack-v5/test-stack-prevent-native-dismiss-single-stack/scenario.md (hunk 1 above)
Comment (LKuchno) [reply apps/src/tests/single-feature-tests/stack-v5/test-stack-prevent-native-dismiss-single-stack/scenario.md]: I kept using Key but scenario is not referring to its format - I will update it also in other scenarios. 
This commit revert changes I did in StackRouteInformation.tsx file and update scenario. 
https://github.com/software-mansion/react-native-screens/pull/4272/changes/292f0b6de212a5d743c76d07f835a2e7571cfa9f
